### PR TITLE
chore: buyget templates add default target + buy rules

### PR DESCRIPTION
**Pull request by riqwan**
what:

- adds a default rule set to buyget templates
- cleanup

### CODE REVIEW

File: packages/admin-next/dashboard/src/lib/client/promotions.ts [hunk 1]
```diff
@@ -81,15 +81,22 @@ async function removePromotionRules(
...
 
-async function listPromotionRules(id: string | null, ruleType: string) {
+async function listPromotionRules(
+  id: string | null,
+  ruleType: string,
+  promotionType?: string
+) {
   return getRequest<PromotionRuleAttributesListRes>(
-    `/admin/promotions/${id}/${ruleType}`
+    `/admin/promotions/${id}/${ruleType}?promotion_type=${promotionType}`
```
Comment (sradevski) [inline packages/admin-next/dashboard/src/lib/client/promotions.ts]: Should we do 
```suggestion
if(promotionType) {
const promoFilter = `promotion_type=${promotionType}`
...
```

Or does passing an empty `promotion_type=` work?

File: packages/admin-next/dashboard/src/routes/promotions/common/edit-rules/components/edit-rules-form/edit-rules-form.tsx [hunk 2]
```diff
@@ -39,65 +37,38 @@ const EditRules = zod.object({
...
-
   const handleFormSubmit = form.handleSubmit(handleSubmit(rulesToRemove))
 
   return (
     <RouteDrawer.Form form={form}>
       <form onSubmit={handleFormSubmit} className="flex h-full flex-col">
         <RouteDrawer.Body>
           <RulesFormField
-            form={form}
+            form={form as any}
```
Comment (sradevski) [inline packages/admin-next/dashboard/src/routes/promotions/common/edit-rules/components/edit-rules-form/edit-rules-form.tsx]: What's the issue with the typings here, maybe we can resolve it without any?

File: packages/admin-next/dashboard/src/routes/promotions/common/edit-rules/components/rules-form-field/rules-form-field.tsx [hunk 3]
```diff
@@ -34,20 +26,84 @@ type RulesFormFieldType<TSchema extends FieldValues> = {
...
+  const { attributes } = usePromotionRuleAttributes(ruleType, formData.type)
+  const { operators } = usePromotionRuleOperators()
+
+  const { fields, append, remove, update, replace } = useFieldArray({
+    control: form.control,
+    name: scope,
+    keyName: scope,
+  })
+
+  const watchType = useWatch({
```
Comment (sradevski) [inline packages/admin-next/dashboard/src/routes/promotions/common/edit-rules/components/rules-form-field/rules-form-field.tsx]: Can we rename this to something more obvious (it sounds like a function now), eg. `ruleTypeValue`

File: packages/admin-next/dashboard/src/routes/promotions/common/edit-rules/components/edit-rules-form/edit-rules-form.tsx (hunk 2 above)
Comment (riqwan) [reply packages/admin-next/dashboard/src/routes/promotions/common/edit-rules/components/edit-rules-form/edit-rules-form.tsx]: This was legacy, forgot to remove it. 🤞🏻 Good catch

File: packages/admin-next/dashboard/src/routes/promotions/common/edit-rules/components/edit-rules-wrapper/edit-rules-wrapper.tsx [hunk 4]
```diff
@@ -60,36 +40,24 @@ export const EditRulesWrapper = ({
...
 
   const handleSubmit = (rulesToRemove?: { id: string }[]) => {
-    return async function (data: { rules: PromotionRuleDTO[] }) {
+    return async function (data: { rules: PromotionRuleResponse[] }) {
       const applicationMethodData: Record<any, any> = {}
       const { rules: allRules = [] } = data
-      const disguisedRulesData = allRules.filter((rule) =>
-        disguisedRules.map((rule) => rule.id).includes(rule.id!)
-      )
+      const diguisedRules = allRules.filter((rule) => rule.disguised)
```
Comment (fPolic) [inline packages/admin-next/dashboard/src/routes/promotions/common/edit-rules/components/edit-rules-wrapper/edit-rules-wrapper.tsx]: nit: typo

**Review by fPolic — COMMENTED** [review]
Not sure if this is directly related to this PR but when I select "By Get" type, I cannot submit request successfully since the rules array is always empty?

**Review by fPolic — APPROVED** [review]
Looks good!

One minor thing we could address before merge is that when you click "clear all" required fields are not reset to default/empty values.

Also, I noticed that we set default values for amount and percentage to 100 which is not great from the UX perspective. Think that we shouldn't set defaults there and use placeholders.

**Comment by riqwan** [comment]
Thanks @fPolic , I'll create a ticket with these changes. I can't seem to easily make the reset work for the select without creating additional bugs. 😅 Ticket - CORE-2272

**Pull request merged**